frontend/src/scenes/feature-flags/featureFlagLogic.ts

```diff
@@ -443,6 +445,7 @@ export const featureFlagLogic = kea<featureFlagLogicType>([
                         surveys: [...(state.surveys || []), newSurvey],
                     }
                 },
+                setId: (state, { id }) => ({ ...state, id }),
```

Comment:
This is not ok, we shouldn't be changing a prop level data, that changes the logic key like so. Rather, the id should be passed in at logic initialisation time as a prop, from which everything else flows downwards, like loading the flag data. (like what we do in bindlogic)

In here, we have to be wary of having a new flag id connected to the old flag id data.

In what case does the id need to be 'schedule' for the featureflaglogic? iiuc, scheduling can only happen on an existing flag, so the regular bindlogic should work perfectly for scheduling as well, without the new 'schedule' id? Am I missing something?

Reply:
Ah, I'm guessing you need it right now to show a new release condition, even for an existing flag. Reasonable to have a different keyed logic there so the existing release conditions don't show up 👍 - wondering if there is a better way. But let's keep it isolated there and use the right logic everywhere else.

Reply:
(ok to revisit, I'm not sure what's the best way here yet, will have a look once back 👀 )

Reply:
I guess I'm not fully appreciating where breaking the assumption might become an issue, but I see that this is becoming a bit of a spaghetti.

I did consider passing the feature flag id as a parameter. The issue is that the success listener also needs this value (to reload the list) - so it would require returning the flag id value from the loader, when the loader should just return the _loaded value_ itself.

I do like reusing the same logic for both, while keeping the state separate by having the logic copied for the `schedule` use case. Any scheduled flag change, additive or not, changes the same model and thus should be possible to express with the same logic.

Also would love to revisit when you're back :) are you okay merging this to keep moving? Not planning to introduce any more changes/mess to the logic before we revisit this.

Reply:
Yeah all good

Reply:
[reminder for self: I suspect the ideal refactor here would be to FeatureFlagReleaseConditions to have an onChange handler that returns filters to you to do whatever with. The form handler just updates, the schedule handler sets is to `scheduledFieldUpdate` - so the only place the 'id=schedule' logic is required is the release condition, which keeps things decently isolated, and the regular flag logic is used everywhere.]